### PR TITLE
Sanitize input between PR deployment workflows.

### PR DESCRIPTION
**Pull request by axisofentropy**

### CODE REVIEW

**Comment by axisofentropy** [comment]
All other projects have merged these fixes so I think it's now safe to say that this PR mitigates the security issue identified by https://github.com/backstage/backstage/pull/21912

Several mitigations are included:
- Only two files are explicitly extracted from the .zip archive, not any.
- `GITHUB_OUTPUT` is used instead of `GITHUB_ENV`, which gets executed.
- Regular expressions sanitize the JSON input from the first workflow.
- The `harden-runner` action now enforces a policy preventing other network connections.
- The permissions of the included token are restricted to only accessing the GHA cache.

Combined, I think these create the most secure GitHub Action Job I've ever seen. I hope we can get this merged and re-enable the preview deployments.

cc @Rugvip @benjdlambert

**Review by Rugvip — APPROVED** [review]
Nice, thank you! 👍 🎉  

Agree there are quite a couple more layers added now which should hopefully keep this secure even if there are individual issues in places. Still a bit of a scary workflow overall though, and we'll be careful with future changes too.

File: .github/workflows/uffizzi-preview.yaml [hunk 1]
```diff
@@ -12,16 +12,23 @@ jobs:
...
+      action: ${{ steps.event.outputs.ACTION }}
     steps:
       - name: Harden Runner
         uses: step-security/harden-runner@eb238b55efaa70779f274895e782ed17c84f2895 # v2.6.1
         with:
-          egress-policy: audit
+          disable-sudo: true
+          egress-policy: block
+          allowed-endpoints: >
+            api.github.com:443
```
[the remark this answers is not in the document]
Comment (axisofentropy) [reply .github/workflows/uffizzi-preview.yaml:31]: This prevents making connections to domains other than GitHub, both to download more executables and upload secrets. Both of these were demonstrated in the associated Proof of Concept.

[the remark this answers is not in the document]
Comment (axisofentropy) [reply .github/workflows/uffizzi-preview.yaml:31]: Could both of those things happen through the GitHub API? Yeah I guess. It does complement the other protections.

**Pull request merged**